src/dmd/expression.d

```diff
@@ -2320,6 +2320,28 @@ extern (C++) abstract class Expression : RootObject
         return type ? 1 : 0; // default modifiable
     }
 
+    /*******************************
+     * Checks the attributes of a function.
+     * Purity (`pure`), safety (`@safe`), no GC allocations(`@nogc`)
+     * and usage of `deprecated` and `@disabled`-ed symbols are checked.
+     *
+     * Params:
+     *  sc = scope of the function
+     *  f  = function to be checked
+     * Returns: `true` if error occur.
+     */
+    final bool checkFunctionAttributes(Scope* sc, FuncDeclaration f)
```

Comment:
I think this would be better as a `private static` function in the `expressionsem.d` module.

Reply:
Done.